### PR TITLE
`matching(X, y)` should print a warning if `y` is a table and `y` only has one column

### PR DESCRIPTION
**Pull request by DilumAluthge**
See https://github.com/alan-turing-institute/MLJ.jl/issues/937 for motivation.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
cc: @Carlos-Trebbau

File: src/model_search.jl [hunk 1]
```diff
@@ -251,11 +251,32 @@ end
...
-matching(X, y, w::AbstractDict) =
-    Checker{true,false,true,scitype(X),scitype(y)}()
+function warn_if_single_column(y)
+    if Tables.istable(y) && (length(Tables.columnnames(y)) == 1)
+        msg = string(
+            "y is a table with only one column. ",
+            "If y is a table, we assume you want to do multi-target modeling.",
+            "If you actually want to do single-target modeling, ",
+            "you need to convert y to a Vector.",
+        )
```
Comment (ablaom) [inline src/model_search.jl:261]: Please enclose variable names in single quotes.

**Review by ablaom — CHANGES_REQUESTED** [review]
Thanks for this, @DilumAluthge . This is a good implementation. 

Can you please make the warning string a global constant, but defined in the same file, close to where you use it. You could call it `WARN_MULTITARGET`, say. And can you please add a `log_test` that references that constant?

File: src/model_search.jl (hunk 1 above)
Comment (DilumAluthge) [reply src/model_search.jl]: Done

**Review by ablaom — APPROVED** [review]
Beautiful.

**Pull request merged**